src/actions/actionsFactory.ts

```diff
@@ -1,40 +1,89 @@
 import { Uri } from "vscode";
-import { IAction, ICommandAction, ActionType, IFileAction, ActionJsonKey } from './interfaces';
-import { CommandAction, FileAction } from './impl';
+import { IAction, ICommandAction, ActionType, IFileAction, ActionJsonKey, ISnippetAction } from './interfaces';
+import { CommandAction, FileAction, SnippetAction } from './impl';
+import _ = require("lodash");
+
+const getNameProp = (fromSettings: boolean): string => {
+    return fromSettings ? "name" : ActionJsonKey.CommandName;
+};
+const getParamsProp = (fromSettings: boolean): string => {
+    return fromSettings ? "params" : ActionJsonKey.CommandParams;
+};
 
 export class ActionsFactory {
-    public static createAction(jsonAction: any) : IAction {
+    public static createAction(jsonAction: any, fromSettings = false): IAction {
         const actionType = jsonAction[ActionJsonKey.ActionType];
         if (!actionType) {
             throw new Error(`${ActionJsonKey.ActionType} is missing`);
         }
         switch (actionType) {
             case ActionType.Command: {
-                    const commandAction: ICommandAction = new CommandAction();
-                    const commandName = jsonAction[ActionJsonKey.CommandName];
-                    if (commandName) {
-                        commandAction.name = commandName;
-                    } else {
-                        throw new Error(`${ActionJsonKey.CommandName} is missing for actionType=${actionType}`);
-                    }
-                    const commandParams = jsonAction[ActionJsonKey.CommandParams];
-                    if (commandParams) {
-                        commandAction.params = commandParams;
-                    }
-                    return commandAction;
+                const commandAction: ICommandAction = new CommandAction();
```

Comment:
please Increase readability be extracting each handler to a separate sub function

```javascript
switch (actionType) {
            case ActionType.Command: {
                return handleCommandAction(args...)
            }
            case ActionType.Snippet: {
                return handleSnippetAcion(args...)
            }
            case ActionType.File: {
                return handleFileAction(args...)
            }
            default:
                throw new Error(`Action with ${ActionJsonKey.ActionType}=${actionType} could not be created from json file`);
        }
```

Reply:
please resolve